telegram/src/main/kotlin/com/github/kotlintelegrambot/network/serialization/adapter/TelegramFileAdapter.kt

```diff
@@ -14,5 +15,6 @@ internal class TelegramFileAdapter : JsonSerializer<TelegramFile> {
         is ByFileId -> context.serialize(src.fileId, String::class.java)
         is ByUrl -> context.serialize(src.url, String::class.java)
         is ByFile -> context.serialize("attach://${src.file.name}")
+        is ByByteArray -> context.serialize("attach://${src.filename!!}")
```

Comment:
Won't this crash if the `ByByteArray` object doesn't have a filename? Should we enforce a filename/provide a fallback for it?

Reply:
This case is covered in [`sendVoice with audio from ByteArray`](https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/blob/46d2e8c603c1379fe9c76d6919a2ba39c0f7d32d/telegram/src/test/kotlin/com/github/kotlintelegrambot/network/apiclient/SendVoiceIT.kt#L89) test, which passes.

Actually, this code line is never called (as well as preceding; it's here only for `when` exhaustiveness), because “attach://<file_attach_name>” is never sent as a JSON field, it's used only with "multipart/form-data" (fallback for the filename is provided [during conversion to `MultipartBody.Part`](https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/blob/46d2e8c603c1379fe9c76d6919a2ba39c0f7d32d/telegram/src/main/kotlin/com/github/kotlintelegrambot/network/multipart/MultipartExtensions.kt#L34)).

Reply:
It's a bit tricky, but I think this code path could be reached from the `sendMediaGroup` operation https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/blob/1773290206d217999c73a88bbeec824ba591b50f/telegram/src/main/kotlin/com/github/kotlintelegrambot/network/multipart/MultipartBodyFactory.kt#L39. If you check the code there we use `Gson` to deserialise "groupable" media which ends up deserialising `TelegramFile`s.

However, we don't support `ByteArray` files in that API yet (although a library client is allowed to pass `ByteArray` files to it), so that would need a bit of extra work. We can leave it as is for now and I'll work on the changes to support this properly for `sendMediaGroup` before the next release of the library is shipped if that sounds good to you.